bin/runtime/src/lib.rs

```diff
@@ -336,13 +338,67 @@ parameter_types! {
     pub const Offset: u32 = 0;
 }
 
+// Choose a subset of all the validators for current era that contains all the
+// reserved nodes. Non reserved ones are chosen in consecutive batches for every session
+fn sample() -> Option<Vec<AccountId>> {
```

Comment:
nit: rename the function as it is not sampling but rather choosing.

Reply:
The name was chosen to be forward compatible ;) But I may change that to `rotate` for example?

Reply:
rotate sounds good